fms/utils/serialization.py

```diff
@@ -405,6 +461,11 @@ def load_state_dict_into_model(
     # 1. Get the adapter from checkpoint sd to fms sd
     adapter = _get_adapter(architecture, source)
 
+    # Prepare the extra_kwargs for the adapter
+    adapter_kwargs = {}
+    if hasattr(model, "config"):
+        adapter_kwargs["model_config"] = dataclasses.asdict(model.config)
```

Comment:
`ModelConfig` already has a method `as_dict` which calls the `asdict` method from `dataclasses`